workspaces/ui-v2/src/optic-components/diffs/render/AskForCommitMessage.tsx

```diff
@@ -29,14 +37,28 @@ export default function AskForCommitMessage(props: { hasChanges: boolean }) {
 
   const pendingEndpointsCount = pendingEndpoints.filter((i) => i.staged).length;
 
-  const handleSave = () => {
-    //@todo decide if every endpoint (changes/pending) gets a batch commit id.
+  const handleSave = async () => {
+    // TODO figure out how to attach a commit message to match id
     const commands = context.simulatedCommands;
-    debugger;
-    console.log(commands);
-    //@todo save them with spectacle
-    // saveChanges(commitMessage!);
-    //@todo redirect to /changes-since/{batchCommitId-whenDiffWasRun}
+    const {
+      data: {
+        applyCommands: { batchCommitId },
+      },
+    } = await spectacleMutator({
+      query: `
+      mutation X($commands: [JSON]) {
+        applyCommands(commands: $commands) {
+          batchCommitId
+        }
+      }
+      `,
+      variables: {
+        commands,
+      },
+    });
+    // If there are no batch commits (first commit) - link to the just created commit
+    history.push(changelogPageRoute.linkTo(lastBatchCommitId || batchCommitId));
+    // @nic TODO we need to trigger a new full page refresh - or trigger a refetch of data through out queries
```

Comment:
Since we redirect + our gql query hasn't retriggered, we can either do a full page refresh (i.e. window.location.push) or figure out a way to programatically retrigger fetches (there's some hacky ways, and some less hacky ways that might require more effort)

Reply:
@devdoshi probably has good thoughts on how we should approach this. 

We definitely want the UI to work without a forced refresh so our example sessions and demo sessions that run in the browser continue functioning as expected. A full refresh here would nuke their state w/o some local storage or other persistence

Reply:
Agreed - we definitely want to not full refresh, however since we use CQRS we inherently need to trigger requeries on all our gql things. 

My initial thoughts are we could expose something from the spectacle provider to "refetch all" / invalidate all queries (which is the most correct since any request could modify any query) - and if that proves challenging (or a lot of work) there are slightly hackier ways we can do this as well